src/lowering/desugar.ml

```diff
@@ -598,8 +598,8 @@ and build_actor at ts (exp_opt : Ir.exp option) self_id es obj_typ =
       let (_dom_sort, dom_fields) = T.as_obj (T.normalize dom) in
       let (_rng_sort, rng_fields) = T.as_obj (T.promote rng) in
       let stab_fields_pre =
-        List.sort T.compare_field
-          (dom_fields @
+        List.sort (fun (r1, tf1) (r2, tf2) -> T.compare_field tf1 tf2)
+          ((List.map (fun tf -> (true, tf)) dom_fields) (* required *) @
```

Comment:
```suggestion
          ((List.map (fun tf -> true, tf) dom_fields) (* required *) @
```

Reply:
I don't like the suggestion - sometime more parens are clearer - OCaml tuple syntax is broken IMO